simplemma/strategies/dictionaries/trie_directory_factory.py

```diff
@@ -0,0 +1,161 @@
+import hashlib
+import logging
+from collections.abc import MutableMapping
+from functools import lru_cache
+from pathlib import Path
+from typing import ByteString, Dict, List, Optional, cast
+
+from marisa_trie import BytesTrie, HUGE_CACHE  # type: ignore[import-not-found]
+from platformdirs import user_cache_dir
+
+from simplemma.strategies.dictionaries.dictionary_factory import (
+    SUPPORTED_LANGUAGES,
+    DictionaryFactory,
+    _load_dictionary_from_disk,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TrieWrapDict(MutableMapping):
```

Comment:
One of the constraints I gave myself was to implement this functionality without requiring changes to the existing code of Simplemma. That's why I didn't modify expected return types of the `DictionaryFactory` for example. Doing so would certainly simplify things, however that's something I didn't want to decide on my own.

It's a `MutableMapping` right now, because a `dict`, what's used when using the `DefaultDictionaryFactory`, is as well. No further reason beyond that.